classes/mutex/PHPRedisMutex.php

```diff
@@ -55,18 +62,34 @@ protected function add($redisAPI, string $key, string $value, int $expire): bool
         }
     }
 
+    /**
+     * {@inheritDoc}
+     *
+     * @param \Redis|\RedisCluster $redis The Redis or RedisCluster connection.
+     */
     protected function evalScript($redis, string $script, int $numkeys, array $arguments)
     {
-        /** @var Redis $redis */
+        // Determine if we need to compress eval arguments.
+        $lzfCompression = false;
```

Comment:
This should be moved into a private method: `hasLzfCompression(Redis $redis): bool`.

Reply:
Will do that. Will not typehint Redis as we might also get a RedisCluster class. Unfortunately there is no inheritance between those two.

Reply:
Will typehint it in the docblock.